src/zarr/codecs/gpu.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+import asyncio
+from dataclasses import dataclass
+from functools import cached_property
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from zarr.abc.codec import BytesBytesCodec
+from zarr.core.common import JSON, parse_named_configuration
+from zarr.registry import register_codec
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+    from typing import Self
+
+    from zarr.core.array_spec import ArraySpec
+    from zarr.core.buffer import Buffer
+
+try:
+    import cupy as cp
+except ImportError:
+    cp = None
+
+try:
+    from nvidia import nvcomp
+except ImportError:
+    nvcomp = None
+
+
+def parse_zstd_level(data: JSON) -> int:
+    if isinstance(data, int):
+        if data >= 23:
+            raise ValueError(f"Value must be less than or equal to 22. Got {data} instead.")
```

Comment:
Ideally nvcomp would do this, but if we're taking it on then we'll need unit tests for the `ValueError` here, the `TypeError` [here]https://github.com/zarr-developers/zarr-python/pull/2863/files#diff-49b79e9bc6f6a9c0f6714c1c1be4b6f4c1539b0afabbb72568b8383ead14a5f0R37) and the `TypeError` [here](https://github.com/zarr-developers/zarr-python/pull/2863/files#diff-49b79e9bc6f6a9c0f6714c1c1be4b6f4c1539b0afabbb72568b8383ead14a5f0R43).